File: py/2466.py
def sinuca(row, n):
    if n > 1:
        newRow = []
        
        for i in range(len(row) - 1):
            if row[i] + row[i + 1] == 0:
                newRow.append(1)
            
            else:
                newRow.append(-1)
                
        return sinuca(newRow, n - 1)
        
    else:
        return row
    
n = int(input())
row = [int(i) for i in input().split()]

row = sinuca(row, n)

if row[0] == 1:
    print('branca')

else:
    print('preta')

File: cpp/2466.cpp
#include <iostream>
#include <vector>

using namespace std;

bool sinuca(vector<int> row, int n)
{
    vector<int> new_row;

    while (n > 1) {
        n--;

        for (int i = 0; i < n; i++) {
            if (row.at(i) + row.at(i + 1) == 0) {
                new_row.push_back(1);
            } else {
                new_row.push_back(-1);
            }
        }

        row.clear();
        row.assign(new_row.begin(), new_row.end());
        new_row.clear();
    }

    return (row.at(0) == 1) ? true : false;
}

int main()
{
    int n, i, j;
    cin >> n;
    vector<int> row;

    for (i = 0; i < n; i++) {
        cin >> j;
        row.push_back(j);
    }

    if (sinuca(row, n)) {
        cout << "branca\n";
    } else {
        cout << "preta\n";
    }

    return 0;
}